nntrainer/src/databuffer.cpp

```diff
@@ -106,6 +116,53 @@ bool DataBuffer::init(int mini_batch, unsigned int train_bufsize,
   return true;
 }
 
+int DataBuffer::init() {
+
+  int status = ML_ERROR_NONE;
+
+  if (!this->class_num) {
+    ml_loge("Error: number of class must be set");
+    for (int i = 0; i < NBUFTYPE; ++i)
+      validation[i] = false;
+    return ML_ERROR_INVALID_PARAMETER;
+  }
+
+  if (!this->input_size) {
+    ml_loge("Error: featuer size must be set");
+    for (int i = 0; i < NBUFTYPE; ++i)
+      validation[i] = false;
+    return ML_ERROR_INVALID_PARAMETER;
+  }
+
+  this->cur_train_bufsize = 0;
+  this->cur_val_bufsize = 0;
+  this->cur_test_bufsize = 0;
+
+  this->train_bufsize = bufsize;
+  this->val_bufsize = bufsize;
+  this->test_bufsize = bufsize;
+
+  if (mini_batch == 0) {
+    ml_loge("Error: mini batch size must be greater than 0");
+    for (int i = 0; i < NBUFTYPE; ++i)
+      validation[i] = false;
+    return ML_ERROR_INVALID_PARAMETER;
+  }
+
+  this->rest_train = max_train;
+  this->rest_val = max_val;
+  this->rest_test = max_test;
+
+  this->train_running = true;
```

Comment:
Q) As far as I know, carry out each step in the order of the train-validation->train->validation...->test(at last). Is it okay if all the step running flags are made true?

Reply:
This is the flag to synchronize with data generation thread. If this flag is false, then the data generation thread will stop there.

Reply:
I got it thanks for the explanation!